### PR TITLE
Cooked-matrix tracker working together with the primary-vertex finder

### PR DESCRIPTION
**Pull request by iouribelikov**
Itself, this PR is already functional.  Please try it, and review the code.
However, the reconstruction workflow will not work with the "--disable-mc" flag until the PR #1963 is merged.

### CODE REVIEW

**Comment by iouribelikov** [comment]
@shahor02 Hi Ruben. I've removed the "WIP" from this PR. In principal, it could be merged now.  
Could you have a look at the changes, please ?

File: Detectors/ITSMFT/ITS/reconstruction/src/CookedTracker.cxx [hunk 1]
```diff
@@ -437,9 +437,9 @@ std::vector<TrackITSExt> CookedTracker::trackInThread(Int_t first, Int_t last)
   seeds.reserve(last - first + 1);
 
   for (auto& vtx : mVertices) {
```
Comment (shahor02) [inline Detectors/ITSMFT/ITS/reconstruction/src/CookedTracker.cxx]: const auto& vtx ?

File: Detectors/ITSMFT/ITS/reconstruction/include/ITSReconstruction/CookedTracker.h [hunk 2]
```diff
@@ -45,14 +46,19 @@ namespace its
...
+  using Vertex = o2::dataformats::Vertex<o2::dataformats::TimeStamp<int>>;
 
  public:
   CookedTracker(Int_t nThreads = 1);
   CookedTracker(const CookedTracker&) = delete;
   CookedTracker& operator=(const CookedTracker& tr) = delete;
   ~CookedTracker() = default;
 
-  void setVertices(std::vector<std::array<Double_t, 3>>& vertices) { mVertices = std::move(vertices); }
+  void setVertices(std::vector<Vertex>& vertices)
```
Comment (shahor02) [inline Detectors/ITSMFT/ITS/reconstruction/include/ITSReconstruction/CookedTracker.h]: Why the tracker needs to own the vertices? Isn't it better just to pass ``const std::vector<Vertex>& vertices`` and assign it to a const pointer?

**Review by shahor02 — COMMENTED** [review]
@iouribelikov Thanks, look fine, just a question below.

**Comment by iouribelikov** [comment]
@shahor02 Thank you for the review, Ruben. Both your suggestions have been implemented now.

**Comment by shahor02** [comment]
@iouribelikov Thanks! Merging.

**Pull request merged**